test/PoolSwapLibrary/mulFraction.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { ethers } from "hardhat"
+import chai from "chai"
+import chaiAsPromised from "chai-as-promised"
+import { PoolSwapLibrary, PoolSwapLibrary__factory } from "../../types"
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+
+chai.use(chaiAsPromised)
+const { expect } = chai
+
+describe("PoolSwapLibrary - mulFraction", () => {
+    let signers: SignerWithAddress[]
+    let library: PoolSwapLibrary
+    beforeEach(async () => {
+        // Deploy the contracts
+        signers = await ethers.getSigners()
+
+        const libraryFactory = (await ethers.getContractFactory(
+            "PoolSwapLibrary",
+            signers[0]
+        )) as PoolSwapLibrary__factory
+
+        library = await libraryFactory.deploy()
+        await library.deployed()
+    })
+    it("should return 0 if the denominator is 0", async () => {
```

Comment:
My preference is still to lean into BDD terminology (solely because we're bothering to use a BDD testing framework in the first place); however, given time constraints on us currently I think it's tolerable for this to be omitted.